### ISSUE TITLE
Switch protocol to long-polling instead of spamming requests

### ISSUE THREAD

**Issue by LPGhatguy** [LPGhatguy/rojo#43]
This would drastically reduce Rojo's syncing latency and reduce the number of HTTP requests that are burned on syncing.

**Comment by Quenty** [comment]
I think you should obscure the long-polling behavior on the Lua plugin side as a normal signal and provide it as a generic API. I've already got code if you're interested.

**Comment by LPGhatguy** [comment]
You should definitely share that code.

When I tried to implement long polling on top of our (pretty terrible) HTTP API, I had a lot of problems making it robust enough.

**Comment by Quenty** [comment]
https://github.com/Quenty/roblox-sync-experiment/blob/master/plugin/src/modules/LongPoll.lua

Long syncing with node.js and Lua!

**Comment by Quenty** [comment]
Note the style of programming is a bit out of date.

**Comment by ShiqiAi** [comment]
that is equivalent to implement "RemoteEvent" ?

**Comment by Quenty** [comment]
Yes, it's an alternative to websockets or some sort of other event notification system instead of regular-interval polling. Basically, it makes a request and the web-server refuses to respond until it receives an event, at which point it fulfills the event. 

https://www.pubnub.com/blog/2014-12-01-http-long-polling/

**Comment by ShiqiAi** [comment]
there are some details I am not sure.
1. when Http request timeout, Roblox may throw an error, then we should reestablish a new connection immediately again right?
2. how to send a new request to server when the old one is still pending? drop long polling connection or just send a new request?

**Comment by Quenty** [comment]
Yes. My implementation does handle Http request timeouts. Long polling is used to receive information from the server only, since the server can't long-poll our plugin. 

To send information you still make Http requests as normal.

**Comment by ShiqiAi** [comment]
Cool~
